test/integration/set_delegating_route_test.cc

```diff
@@ -0,0 +1,97 @@
+#include "test/integration/http_integration.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+class SetDelegatingRouteTest : public testing::TestWithParam<Network::Address::IpVersion>,
```

Comment:
optional nit, I think this would be fine folded into another integration test (intergration_test.cc would be fine, or http2_integration_test.cc if you want to keep it H2) so we have one less thing to link.)

Reply:
One less thing to link is nice, done